client/src/manager.rs

```diff
@@ -208,32 +225,19 @@ impl Manager {
 		// --- END ---
```

Comment:
Maybe we should move this into the test `should_calculate_proof`, instead of doing sanity checks

Reply:
Sure! Makes sense, the snippet only runs on unoptimized builds